common/serde-reflection/src/de.rs

```diff
@@ -4,23 +4,32 @@
 use crate::{
     error::{Error, Result},
     format::{ContainerFormat, ContainerFormatEntry, Format, FormatHolder, Named, VariantFormat},
-    trace::Tracer,
+    trace::{Records, Tracer},
 };
 use serde::de::{self, DeserializeSeed, IntoDeserializer, Visitor};
 use std::collections::BTreeMap;
 
-pub(crate) struct Deserializer<'a> {
+pub(crate) struct Deserializer<'de, 'a> {
     tracer: &'a mut Tracer,
+    records: &'de Records,
```

Comment:
This was not needed for Serializer, so I don't understand why it is needed here.

Reply:
(part 2) In the case of Deserializer, we want to be able to borrow bytes from the records (done in the next commit).
Here the lifetime of Records must not "shrink" but stay equal to the lifetime 'de in the trait `serde::de::Deserializer<'de>`. By separating, Records from Tracer we are able to make it read-only in the Deserializer, thus enabling both precise lifetime computation and non-exclusive borrowing.